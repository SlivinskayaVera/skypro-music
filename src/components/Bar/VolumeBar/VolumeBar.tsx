import classNames from "classnames";
import styles from "../Bar.module.css";
import { SVG } from "@/components/SVGImage/SVGImage";
import { useRef, useState } from "react";

type VolumeBarType = {
  audioRef: HTMLAudioElement | null;
};

export function VolumeBar({ audioRef }: VolumeBarType) {
  const [volume, setVolume] = useState<string | null>();
  const volumeRef = useRef<HTMLInputElement | null>(null);

  function handleVolumeChange() {
    if (!audioRef || !volume) return;
    audioRef.volume = +volume;
  }

  return (
    <div className={styles.barVolumeBlock}>
      <div className={styles.volumeContent}>
        <div className={styles.volumeImage}>
          <SVG className={styles.volumeSvg} url="volume" />
        </div>
        <div className={classNames(styles.volumeProgress, styles._btn)}>
          <input
            ref={volumeRef}
            onChange={() => {
              setVolume(volumeRef.current?.value), handleVolumeChange();
            }}
            className={classNames(styles.barPlayerProgress, styles._btn)}
            type="range"
            name="range"
            min="0"
            max="1"
            step={0.0000000001}
          />
        </div>
      </div>
    </div>
  );
}
