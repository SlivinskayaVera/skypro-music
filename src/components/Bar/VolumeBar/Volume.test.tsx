import renderer from "react-test-renderer";
import { VolumeBar } from "./VolumeBar";

describe("Volume bar", () => {
  test("correct display", () => {
    const component = renderer.create(<VolumeBar audioRef={null} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
