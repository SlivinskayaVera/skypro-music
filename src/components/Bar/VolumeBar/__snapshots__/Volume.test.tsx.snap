// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Volume bar correct display 1`] = `
<div
  className="barVolumeBlock"
>
  <div
    className="volumeContent"
  >
    <div
      className="volumeImage"
    >
      <svg
        className="volumeSvg"
      >
        <use
          xlinkHref="/img/icon/sprite.svg#icon-volume"
        />
      </svg>
    </div>
    <div
      className="volumeProgress _btn"
    >
      <input
        className="barPlayerProgress _btn"
        max="1"
        min="0"
        name="range"
        onChange={[Function]}
        step={1e-10}
        type="range"
      />
    </div>
  </div>
</div>
`;
